core/src/consensus/light_client/verification.rs

```diff
@@ -14,72 +14,131 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use ccrypto::blake256;
 use ctypes::CompactValidatorSet;
 use merkle_trie::proof::{verify, CryptoProof, CryptoProofUnit};
+use super::{LightHeader, CoreHeader, Seal};
+use ckey::{verify_schnorr};
+use primitives::{H256, Bytes};
 
-use super::LightHeader;
-use super::Seal;
+pub fn verify_signature(block_hash: H256, vset: &CompactValidatorSet, seal: &Seal) -> bool {
+    let n = vset.0.len();
+
+    for (index, sign) in &seal.0 {
+        if *index >= n {
+            return false;
+        }
+        let (pk, _) = vset.0[*index];
+        match verify_schnorr(&pk, &sign, &block_hash) {
+            Ok(true) => (),
+            Ok(false) => return false,
+            _ => return false,
+        };
+    }
+    true
+}
+
+pub fn verify_quorum(vset: &CompactValidatorSet, seal: &Seal) -> bool {
+    // Note that invalid index would already be rejcted in verify_signature()
+    let total_delegation: u64 = vset.0.iter().map(|b| b.1).sum();
+    let voted_delegation: u64 = seal.0.iter().map(|b| vset.0[b.0].1).sum();
+    if total_delegation * 2 >= voted_delegation * 3 {
+        return false
+    }
+    return true;
```

Comment:
I came up with a thought that this part should be moved to the stake module in the future.
Anyway, you don't need to change it now.

Reply:
Exactly which part can be customized by the module?

Reply:
I heard that other developers who designing the module system are trying to move the staking logic to a module. Some staking rules may not have a delegation, so I thought this code can be changed after the module team's decision.

Since it is not fixed yet, let's recheck this code after some decisions are created in the module system.